e2e/cluster.go

```diff
@@ -147,8 +148,29 @@ func hash(cluster *armcontainerservice.ManagedCluster) string {
 	return hexHash[:5]
 }
 
+func getOrCreateCluster(ctx context.Context, t *testing.T, cluster *armcontainerservice.ManagedCluster) (*armcontainerservice.ManagedCluster, error) {
+	existingCluster, err := config.Azure.AKS.Get(ctx, config.ResourceGroupName, *cluster.Name, nil)
+	var azErr *azcore.ResponseError
+	if errors.As(err, &azErr) && azErr.StatusCode == 404 {
+		return createNewAKSClusterWithRetry(ctx, t, cluster)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster %q: %w", *cluster.Name, err)
+	}
+	switch *existingCluster.Properties.ProvisioningState {
+	case "Succeeded":
+		return &existingCluster.ManagedCluster, nil
+	case "Creating", "Updating":
```

Comment:
should we add "Upgrading" here as well for completeness? granted I don't think we'd ever actually be in that situation unless AKS force-upgrades us